src/app/dashboard/game.service.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { GameService } from './game.service';
+import { TestBed } from '@angular/core/testing';
+import Konva from 'konva';
+import { HEROES } from '../heroes/mock-heroes';
+
+describe('Game service', () => {
+  let service: GameService;
+  const layer = new Konva.Layer();
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(GameService);
+  });
+
+  it('Init game service', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('Draw a konva ring', () => {
+    expect(service.drawRing(0, 0, 50, 'red')).toEqual(jasmine.any(Konva.Ring));
```

Comment:
The purpose of writing tests is so that another developer can't come by and accidently break your functionality.  Testing that drawRing will return a type of Ring doesn't protect you at all.   It is too obvious and not prone to being messed up.  

Same with tests below.

If there is no if/else inside drawRing that could be screwed up then you don't need to test it.

Reply:
Thanks for your comment. I do not have much experience in testing. I will improve myself in it.